build.sbt

```diff
@@ -112,7 +112,7 @@ libraryDependencies ++= Seq(
 dependencyOverrides += "com.google.guava" % "guava" % "16.0.1"
 dependencyOverrides += "commons-codec" % "commons-codec" % "1.6"
 dependencyOverrides += "commons-logging" % "commons-logging" % "1.2"
-dependencyOverrides += "io.netty" % "netty-all" % "4.0.42.Final"
+dependencyOverrides += "io.netty" % "netty-all" % "4.1.17.Final"
```

Comment:
Seems we should update this per https://issues.apache.org/jira/browse/SPARK-22324. Otherwise, it hits an error as below:

```
[info] Exception encountered when attempting to run a suite with class name: com.hortonworks.spark.sql.hive.llap.TestLlapQueryExecutionListener *** ABORTED *** (774 milliseconds)
[info]   java.lang.NoSuchMethodError: io.netty.buffer.PooledByteBufAllocator.metric()Lio/netty/buffer/PooledByteBufAllocatorMetric;
[info]   at org.apache.spark.network.util.NettyMemoryMetrics.registerMetrics(NettyMemoryMetrics.java:80)
[info]   at org.apache.spark.network.util.NettyMemoryMetrics.<init>(NettyMemoryMetrics.java:76)
[info]   at org.apache.spark.network.client.TransportClientFactory.<init>(TransportClientFactory.java:109)
[info]   at org.apache.spark.network.TransportContext.createClientFactory(TransportContext.java:99)
[info]   at org.apache.spark.rpc.netty.NettyRpcEnv.<init>(NettyRpcEnv.scala:71)
[info]   at org.apache.spark.rpc.netty.NettyRpcEnvFactory.create(NettyRpcEnv.scala:461)
[info]   at org.apache.spark.rpc.RpcEnv$.create(RpcEnv.scala:57)
[info]   at org.apache.spark.SparkEnv$.create(SparkEnv.scala:249)
[info]   at org.apache.spark.SparkEnv$.createDriverEnv(SparkEnv.scala:175)
[info]   at org.apache.spark.SparkContext.createSparkEnv(SparkContext.scala:256)
```

by missing methods from netty version mismatch.